SWLOR.Game.Server/Native/ResolveAttackRoll.cs

```diff
@@ -24,7 +28,366 @@ public static void RegisterHook()
         [UnmanagedCallersOnly]
         private static void OnResolveAttackRoll(void* thisPtr, void* pTarget)
         {
-            Console.WriteLine("Running OnResolveAttackRoll");
+            /*
+             * Custom attack logic for SWLOR.  This is done as a skill vs skill opposed roll, so most default NWN logic
+             * does not apply.
+             * 
+             * The following default NWN functions don't exist in this engine.
+             * - Miss on 1
+             * - Hit on 20
+             * - Parry
+             * - Coup de Grace
+             * - Sneak Attack (/Death Attack)
+             * 
+             * Armor Class doesn't exist, and non-creature objects are hit automatically. 
+             * Critical hits come from beating the opposed roll by 30 or more.  Crit immunity applies as normal.
+             * 
+             * A stunned or incapable defender does not add their skill to the opposed roll.  
+             * 
+             */
+
+            Log.Write(LogGroup.Attack, "Running OnResolveAttackRoll");
+            var targetObject = CNWSObject.FromPointer(pTarget);
+            if (targetObject == null)
+                return;
+
+            CNWSCreature attacker = CNWSCreature.FromPointer(thisPtr);
+            CNWSCreatureStats attackerStats = attacker.m_pStats;
+            CNWSCombatRound pCombatRound = attacker.m_pcCombatRound;
+
+            Log.Write(LogGroup.Attack, "Attacker: " + attacker.GetFirstName().GetSimple(0) + ", defender " + targetObject.GetFirstName().GetSimple(0));
```

Comment:
The logging you've got in here is fine for now but we probably need to add a way to indicate which logs should be disabled for production use. For example, we'll want all of the Property logs to write regardless of environment but this log information here is going to add a lot of overhead with 40+ people engaging in combat. Only mentioning it here as a reminder, no need for you to make any adjustments yet.

Reply:
Yes, absolutely.  Most of these are debug logging.  I didn't see an option to specify debug vs production - but I figure we'll add that before we launch.  During testing, no reason not to have debug on.